src/main/game/Mark.java

```diff
@@ -0,0 +1,7 @@
+package main.game;
+
+public enum Mark {
+    X,
+    O,
+    EMPTY
+}
```

Comment:
👏

Reply:
Nice! Is EMPTY a legitimate `Mark` though? You can't play with it, right?
Java has new type express the difference between "here is something" and "here is nothing, go away!".
Its called `Optional<T>` ([docs](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html)).

Your grid would then become..... List<Optional<Mark>>.
It might be fun to look into it.